### PR TITLE
Fix Mobile.CurrentMovementType vertical handling

### LINKED ISSUE #16384: Infantry displays move anim on paradrop
Regression from #15675.

Fortunately an easy fix, I'll file a PR shortly.

### PR DESCRIPTION
**Pull request by reaperrr**
Fixes #16384.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Mobile.cs [hunk 1]
```diff
@@ -224,9 +224,12 @@ protected override void Created(Actor self)
 		void ITick.Tick(Actor self)
 		{
 			var newMovementTypes = MovementType.None;
-			if (oldPos != CenterPosition)
+			if ((oldPos - CenterPosition).HorizontalLengthSquared != 0)
 				newMovementTypes |= MovementType.Horizontal;
 
+			if ((oldPos - CenterPosition).VerticalLengthSquared != 0)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: IMO `oldPos.Z != CenterPosition.Z` would be clearer, but this isn't a blocker.

**Review by pchote — DISMISSED** [review]

**Comment by reaperrr** [comment]
Updated.

**Comment by reaperrr** [comment]
@abcdefg30 @matjaeck Can one of you confirm the fix?

**Review by ghost — APPROVED** [review]
I tested this already but apparently forgot to submit the review. Tested again, can confirm the bug and fix.

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/9a93466641d809b64af17a15b978e7c4844535bb...937b4e1696fdb7a67837dcfad9704810a7dfcfaa

**Pull request merged**